src/main/resources/db/migration/0300/V320__AdHocPlotObservations.sql

```diff
@@ -0,0 +1,28 @@
+CREATE TABLE tracking.observation_types (
+    id INTEGER PRIMARY KEY,
+    name TEXT NOT NULL UNIQUE
+);
+
+INSERT INTO tracking.observation_types (id, name)
+VALUES (1, 'Monitoring'),
+       (2, 'Biomass Measurements')
+ON CONFLICT (id) DO UPDATE SET name = excluded.name;
+
+ALTER TABLE tracking.observations ADD COLUMN is_ad_hoc BOOLEAN;
+
+UPDATE tracking.observations
+SET is_ad_hoc = false
+WHERE is_ad_hoc IS NULL;
+
+ALTER TABLE tracking.observations ALTER COLUMN is_ad_hoc SET NOT NULL;
+
+ALTER TABLE tracking.observations ADD COLUMN observation_type_id INTEGER REFERENCES tracking.observation_types;
+
+UPDATE tracking.observations
+SET observation_type_id = 1
+WHERE observation_type_id IS NULL;
+
+ALTER TABLE tracking.observations ALTER COLUMN observation_type_id SET NOT NULL;
+
+ALTER TABLE tracking.monitoring_plots ALTER COLUMN planting_subzone_id DROP NOT NULL;
+ALTER TABLE tracking.observation_plots ALTER COLUMN monitoring_plot_history_id DROP NOT NULL;
```

Comment:
I had to add these and plumb their nullability through the pre-existing code. There isn't any mention of subzones in the PRD, and it seems like these plots need to be able to be created before any subzones exist. I also opted to allow the monitoring plot history to be optional, since creation of ad hoc observations will be a single event.